check-default-variables.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+#
+# pre-commit hook:
+# Check for Habitat default variables in a plan.sh file
+#
+# Authors:
+# - Mike Fiedler <miketheman@gmail.com>
+
+required_variables=(
```

Comment:
We might as well add everything that's _Required_ on https://www.habitat.sh/docs/reference/plan-syntax/#basic-settings. Not everything in this list is actually required for Habitat to work, but they should be required for core plans.